rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,571 @@
+
+# Guide-level explanation
+The Glides trainsheet functionality hews closely to what the paper-based trainsheets provide, along with some shortcuts for common actions.
+
+- sign an operator in, marking them fit-for-duty
+- add a trip (1-way from the current location, 1-way arriving at the current location, or shortcut: round-trip from the current location)
+- drop a trip (with a reason)
+- add/change/remove train cars to a trip
+- add/change/remove the operators assigned to a car/trip
+- set a new departure time for a trip
+- shortcut: modifying multiple trips together to manage headways
+- shortcut: removing an operator from multiple trips (due to illness, for example)
+
+These actions are always taken by a logged-in user (generally an inspector).
+
+There are additional automated processes which keep the schedule up-to-date:
+- importing date/schedule mappings
+- importing runs
+- importing scheduled trips
+
+The stream of events proposed by this RFC serves two purposes:
+- primarily, improve the rider-facing predictions produced by RTR, especially for departure times at a terminals, which currently do not have departure predictions at all, and
+- secondarily, other internal operations needs such as Lean or LAMP
+
+The goal of this RFC is to provide an interface which represents all data currently (2023H1) in Glides. While some additional flexibility is included, features outside the current scope of Glides are also outside the scope of these events and this RFC. 
+
+## CloudEvents
+All events will be in the [CloudEvents v1.0.2](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md) format (or later), using the JSON encoding. The event types will be under the `com.mbta.ctd.glides` event namespace, and will be in the past-tense (i.e. `trip-added` rather than `add-trip`).
+
+## Kinesis
+Events will be written to a Kinesis stream. Each Glides environment (`dev`, `dev-green`, and `prod`) will have a separate stream for their events, named `ctd-glides-<environment>`. 
+
+The partition key will be a hash of the station at which the inspector is working and the inspector's identity, which ensures that multiple events from a single inspector are ordered correctly if the messages are distributed across multiple shards. 
+As the stream will include potential PII (operator badge information) it should be configured as encrypted-at-rest. 
+
+Multiple events can be included in a single Kinesis record, by wrapping them in a JSON array. This is an optimization for improving write speeds: further steps in the Kinesis pipeline may break up or rearrange these arrays. If multiple events are part of the same user action, the [Batched][#Batched] event should be used to combine them.
+
+## Value types
+
+### Author
+Each message will include an `author` key, indicating who performed the given intervention.
+
+- `username` (string): the username of the logged-in user
```

Comment:
The value I see in the UI is the username:
<img width="147" alt="Screenshot 2023-02-02 at 8 58 20 AM" src="https://user-images.githubusercontent.com/214921/216344533-a9047c72-fa0d-45e9-9c51-6d5c42afa2ef.png">
Is Glides doing some filtering in the UI to remove the domain?

Reply:
For actual inspectors, we have their name and badge number from HR/CMS, so that's the information that we show there. For users from CTD, we don't have info from HR/CMS, so we calculate someone's name as their email address with a possible `@mbta.com` suffix removed.

Reply:
Makes sense, I'll change it to e-mail.